src/Domain/CoordinateMaps/Frustum.cpp

```diff
@@ -241,12 +259,16 @@ std::optional<std::array<double, 3>> Frustum::inverse(
         coord /= abs(coord);
       }
     }
-    logical_coords = RootFinder::gsl_multiroot(
-        rootfunction, logical_coords, absolute_tolerance, maximum_iterations,
-        relative_tolerance, verbosity, max_absolute_tolerance, method,
-        condition);
+    try {
+      logical_coords = RootFinder::gsl_multiroot(
+          rootfunction, logical_coords, absolute_tolerance, maximum_iterations,
+          relative_tolerance, verbosity, max_absolute_tolerance, method,
+          condition);
+
+    } catch (const convergence_error& e) {
+      return std::nullopt;
```

Comment:
Do you understand why code cov claims this isn't covered? Did you add a `std::cout` here to verify the failing example actually fails this check and not another? If it's a codecov issue that's fine, just want to make sure.